tests/plugins/uploadwidget/uploadwidget.js

```diff
@@ -156,6 +156,59 @@
 			} );
 		},
 
+		// (#1454)
+		'test abort': function() {
+			var bot = this.editorBot,
+				editor = bot.editor;
+
+			var stub = sinon.stub().returns( true );
+
+			addTestUploadWidget( editor, 'testuploadwidget', {
+				onAbort: stub
+			} );
+
+			bot.setData( '', function() {
+				pasteFiles( editor, [ bender.tools.getTestPngFile() ] );
+
+				var loader = editor.uploadRepository.loaders[ 0 ];
+
+				// Remove widget from the DOM.
+				editor.widgets.destroyAll();
+				editor.editable().findOne( '[data-cke-upload-id="' + loader.id + '"]' ).remove();
+
+				loader.changeStatus( 'abort' );
+
+				assert.isTrue( stub.calledOnce );
+			} );
+		},
+
+		// (#1454)
+		'test abort can be called only once': function() {
```

Comment:
Is this test realistic? Is such situation possible to happen? According to our docs, `abort` event is fired only once for every loader.

In fact this test uses private method, `changeStatus`, which does not prevent double setting the same status; yet public `abort` method does it.

Reply:
Of course it could be fixed with slight change to the listener by using `once` instead of `on`.